comictaggerlib/taggerwindow.py

```diff
@@ -1201,26 +1203,35 @@ def commit_metadata(self) -> None:
                 self.clear_dirty_flag()
                 self.update_info_box()
                 self.update_menus()
-            self.fileSelectionList.update_current_row()
 
-            self.metadata = self.comic_archive.read_metadata(self.load_data_style)
+                # Only try to read if write was successful
+                self.metadata, error = self.overlay_ca_read_style(self.load_data_styles, self.comic_archive)
+                if error is not None:
+                    QtWidgets.QMessageBox.warning(
+                        self,
+                        "Read Failed!",
+                        f"One or more of the read styles failed to load for {self.comic_archive.path}, check log for details",
+                    )
+                    logger.error("Failed to load metadata for %s: %s", self.ca.path, error)
+
+            self.fileSelectionList.update_current_row()
             self.update_ui_for_archive()
         else:
             QtWidgets.QMessageBox.information(self, "Whoops!", "No data to commit!")
 
-    def set_load_data_style(self, s: str) -> None:
+    def set_load_data_style(self, load_data_styles: list[str]) -> None:
         if self.dirty_flag_verification(
-            "Change Tag Read Style", "If you change read tag style now, data in the form will be lost.  Are you sure?"
+            "Change Tag Read Style",
+            "If you change read tag style(s) now, data in the form will be lost.  Are you sure?",
         ):
-            self.load_data_style = self.cbLoadDataStyle.itemData(s)
-            self.config[0].internal__load_data_style = self.load_data_style
```

Comment:
Why was this removed?

Reply:
Reverted.

It was because the `self.config[0].internal__load_data_style` is only used when saving and it had to be unreversed and so was set on `closeEvent` it didn't need to be set. Thinking on it, that may have caused issues with the settings window saving.